src/os/bluestore/BlueStore.cc

```diff
@@ -1640,6 +1640,31 @@ int BlueStore::_setup_block_symlink_or_file(
 	  VOID_TEMP_FAILURE_RETRY(::close(fd));
 	  return r;
 	}
+#ifdef HAVE_POSIX_FALLOCATE
+	r = ::posix_fallocate(fd, 0, size);
+	if (r < 0) {
+	  r = -errno;
+	  derr << __func__ << " failed to prefallocate " << name << " file to "
+	       << size << ": " << cpp_strerror(r) << dendl;
+	  VOID_TEMP_FAILURE_RETRY(::close(fd));
+	  return r;
+	}
+#else
+	char data[1024*128];
```

Comment:
@majianpeng i agree. but file backend for bluestore is mostly for testing usage. imho, i am not sure if we want to complicate the code to avoid the ENOSPAC in testing env.